crates/re_types_registry/src/components.rs

```diff
@@ -0,0 +1,509 @@
+// DO NOT EDIT! This file was auto-generated by crates/re_types_builder/src/codegen/rust/registry.rs
+use re_types_core::ComponentName;
+
+/// Information about a Rerun [`component`](re_types_core::Component), generated by codegen.
+pub struct ComponentInfo {
+    /// Markdown docstring for the component.
+    pub docstring_md: &'static str,
+}
+
+/// Runtime "reflection" about components.
+pub type ComponentRegistry = nohash_hasher::IntMap<ComponentName, ComponentInfo>;
+
+/// Markdown docs for each component.
+
+pub fn registry() -> &'static ComponentRegistry {
```

Comment:
Related to the above comment: what makes this particular problematic is that before I was able to add new types without modifying rerun crates, now I can't, this is quite the regression imho. We should design this as a registry that is owned by App and populated by this method (but can also be populated from the outside).

Similarly, we should also move the population of the placeholder (re_viewer/src/component_deafults == component::default()) registry here which is currently a generated file that is part of re_viewer.
With those changes in place, only re_viewer needs to depend on this crate which then avoids stacking our crate hierarchy higher, keeping it flat.
Alternatively we could also consider just adding it to re_viewer directly, just like `component_defaults` - I went that way because it kept things simpler, but there's pros and cons withi that.

Reply:
`re_types_registry` only depends on `re_types_core` - it has a very shallow dependency. It intentionally does NOT depend on `re_types` directly (but does indirectly via codegen).